oat-sa/tao-core
# Fix/act 355/missing events

Related to: https://oat-sa.atlassian.net/browse/ACT-355
Added LockService to prevent worker from processing not logged task.
To test:
1) Configure Queue to use rds `sudo -u www-data php index.php 'oat\taoTaskQueue\scripts\tools\InitializeQueue' --broker=rds --persistence=default --receive=10`
2) Configure LockService to use `RedisStore` `sudo -u www-data php index.php 'oat\tao\scripts\install\SetUpLockService' -p redis` (redis - name of redis persistence)
3) Run worker `sudo -u www-data php index.php 'oat\taoTaskQueue\scripts\tools\RunWorker'`
  3.1) To check in code put brakepoints:
`\oat\tao\model\taskQueue\Queue::enqueue` on  `$isEnqueued = $this->getBroker()->push($task);`
`\oat\tao\model\taskQueue\Queue::dequeue` on `if ($this->canDequeueTask($task)) {`
  3.2) To check without code - do nothing.
4) Run any action that will goes to queue. For example export qti test.
4.1) Will be triggered first brakepoint. Second brakepoint will be triggered only after unpausing first.
4.2) Task completed.
